File: templates/justfile.java
# {{PROJECT_NAME}} - {{PROJECT_DESCRIPTION}}
# Java application (Gradle)

default:
    @just --list

# === Standard Vocabulary ===

build:
    ./gradlew build

test:
    ./gradlew test

dev:
    ./gradlew bootRun

preflight:
    @echo "Running preflight checks..."
    ./gradlew check
    ./gradlew build
    @echo "Preflight complete!"

clean:
    ./gradlew clean
    rm -rf build/

# === Project Specific ===

deps:
    ./gradlew dependencies

lint:
    ./gradlew checkstyleMain checkstyleTest

fmt:
    ./gradlew spotlessApply

# Run with debug
debug:
    ./gradlew bootRun --debug-jvm

# Build fat jar
jar:
    ./gradlew bootJar

File: templates/justfile.go
# {{PROJECT_NAME}} - {{PROJECT_DESCRIPTION}}
# Go application

default:
    @just --list

# === Standard Vocabulary ===

build:
    go build -o bin/ ./...

test:
    go test ./...

dev:
    go run .

preflight:
    @echo "Running preflight checks..."
    go vet ./...
    go test ./...
    go build -o bin/ ./...
    @echo "Preflight complete!"

clean:
    rm -rf bin/
    go clean -cache

# === Project Specific ===

deps:
    go mod download
    go mod tidy

lint:
    golangci-lint run

fmt:
    gofmt -w .

# Run with race detector
test-race:
    go test -race ./...

# Generate code (if applicable)
generate:
    go generate ./...
